server/server.go

```diff
@@ -362,6 +362,8 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if status >= 400 {
 			DefaultErrorFunc(w, r, status)
 		}
+
+		w.WriteHeader(status)
```

Comment:
You can't be so sure. Considering `DefaultErrorFunc` also calls `w.WriteHeader`.